flox/core.py

```diff
@@ -1293,14 +1293,7 @@ def dask_groupby_agg(
         name=f"{name}-chunk-{token}",
     )
 
-    if expected_groups is None:
-        if is_duck_dask_array(by_input):
-            expected_groups = None
-        else:
-            expected_groups = _get_expected_groups(by_input, sort=sort)
-    group_chunks: tuple[tuple[Union[int, float], ...]] = (
-        (len(expected_groups),) if expected_groups is not None else (np.nan,),
-    )
+    group_chunks: tuple[tuple[Union[int, float], ...]]
```

Comment:
@Illviljan the only float value that is valid here is np.nan. Is there a way to narrow this to just np.nan? A simple `Literal[np.nan]` does not work.

Reply:
yeah, I am sad about that as well. :( 

Thought a little about this in https://github.com/dask/dask/pull/9255 but came to the conclusion that adding float with a reminder comment was the most practical solution. Here's a relevant typing thread: https://github.com/python/typing/issues/1160

Reply:
ah too bad. Thanks!